exporters/otlp/otlplog/otlploggrpc/client.go

```diff
@@ -3,12 +3,144 @@
 
 package otlploggrpc // import "go.opentelemetry.io/otel/exporters/otlp/otlplog/otlploggrpc"
 
+import (
+	"time"
+
+	"google.golang.org/genproto/googleapis/rpc/errdetails"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/backoff"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/encoding/gzip"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+
+	"go.opentelemetry.io/otel/exporters/otlp/otlplog/otlploggrpc/internal/retry"
+	collogpb "go.opentelemetry.io/proto/otlp/collector/logs/v1"
+)
+
+// The methods of this type are not expected to be called concurrently.
 type client struct {
-	// TODO: implement.
+	metadata      metadata.MD
+	exportTimeout time.Duration
+	requestFunc   retry.RequestFunc
+
+	// ourConn keeps track of where conn was created: true if created here in
+	// NewClient, or false if passed with an option. This is important on
+	// Shutdown as conn should only be closed if we created it. Otherwise,
+	// it is up to the processes that passed conn to close it.
+	ourConn bool
+	conn    *grpc.ClientConn
+	lsc     collogpb.LogsServiceClient
 }
 
+// Used for testing.
+var newGRPCClient = grpc.NewClient
```

Comment:
Technically, this is not a client but a method. How about appending `fn`?

```suggestion
var newGRPCClientFn = grpc.NewClient
```